### PR TITLE
Docstring + minor updates

### PR DESCRIPTION
**Pull request by samuro95**
In this PR
- update docstrings of several functions and example files.
- enable the plotting of the metrics in the Unfodled case: see examples/demo_vanilla_unfolded.py
- Add the "drunet" dataset for training on a large dataset.
- Debug Deep Equilibrium 

### Checks to be done before submitting your PR
- [ ] `python3 -m pytest tests/` runs successfully.
- [ ] `black .` runs successfully.
- [ ] `make html` runs successfully (in the `docs/` directory).
- [ ] Updated docstrings related to the changes (as applicable).
- [ ] Added an entry to the [CHANGELOG.md](../../CHANGELOG.md) (as applicable) (to come).

### CODE REVIEW

**Review by matthieutrs — APPROVED** [review]

File: deepinv/optim/fixed_point.py [hunk 1]
```diff
@@ -116,16 +116,22 @@ def forward(self, *args, **kwargs):
         Since the prior and parameters (stepsize, regularisation parameter, etc.) can change at each iteration,
         the prior and parameters are updated before each call to the iterator.
 
+        :param bool compute_metrics: if True, the metrics are computed along the iterations. Default: False.
         :param args: optional arguments for the iterator.
         :param kwargs: optional keyword arguments for the iterator.
-        :return: the fixed-point.
+        :return tuple: `(x,metrics)` with `x` the fixed-point solution and
```
Comment (tachella) [inline deepinv/optim/fixed_point.py:122]: missing double ``

File: deepinv/unfolded/deep_equilibrium.py [hunk 2]
```diff
@@ -9,50 +9,56 @@
...
+    \label{eq:fixed_point}
+    \tag{1}
+    v = \left(\frac{\partial \operatorname{FixedPoint}(x^\star)}{\partial x^\star} \right )^T v + u.
+    \end{equation}
 
+    where :math:`u` is the incoming gradient from the backward pass, and :math:`x^\star` is the equilibrium point of the forward pass.
 
-    where :math:`\operatorname{step}_f`, :math:`\operatorname{step}_g` as well as the external parameters can be either learnable modules or
-    proximal / gradient steps.
+    See `<http://implicit-layers-tutorial.org/deep_equilibrium_models/>` for more details.
```
Comment (tachella) [inline deepinv/unfolded/deep_equilibrium.py:27]: an _ is missing after the link

File: deepinv/unfolded/deep_equilibrium.py [hunk 3]
```diff
@@ -9,50 +9,56 @@
...
-    These algorithms take the following form (see :meth:`deepinv.unfolded`):
+    The backward pass is performed using fixed point iterations to find solutions of the fixed-point equation
 
     .. math::
-        \begin{aligned}
-        z_{k+1} &= \operatorname{step}_f(x_k, z_k, y, A, \lambda, \gamma, ...)\\
-        x_{k+1} &= \operatorname{step}_g(x_k, z_k, y, A, \sigma, ...)
-        \end{aligned}
+    \begin{equation}
+    \label{eq:fixed_point}
```
Comment (tachella) [inline deepinv/unfolded/deep_equilibrium.py:20]: Is this well indented?

File: deepinv/unfolded/deep_equilibrium.py [hunk 4]
```diff
@@ -9,50 +9,56 @@
...
         r"""
-        Run the algorithm on the input `y` and physics `physics`.
+        The forward pass of the DEQ algorithm. Compared to :class:`deepinv.unfolded.BaseUnfold`, the backward algorithm is performed using fixed point iterations.
 
         :param torch.Tensor y: Input tensor.
         :param deepinv.physics physics: Physics object.
-        :return: Output torch.Tensor.
+        :param torch.Tensor x_gt: (optional) ground truth image, for plotting the PSNR across optim iterations.
+        :param bool compute_metrics: whether to compute the metrics or not. Default: `False`.
+        :return: If `compute_metrics` is False,  returns (torch.Tensor) the output of the algorithm.
```
Comment (tachella) [inline deepinv/unfolded/deep_equilibrium.py:44]: double `` missing

File: examples/unfolded/demo_DEQ.py [hunk 5]
```diff
@@ -2,7 +2,9 @@
 Deep Equilibrium (DEQ) algorithms for image deblurring
 ====================================================================================================
 
-This example shows you how to use DEQ to solve a deblurring problem.
+This a toy example to show you how to use DEQ to solve a deblurring problem. 
+Note that this is a small dataset for training. For optimal results, use a larger dataset.
+For vializing the training, you can use Weight&Bias (wandb) by setting `wandb_vis=True`.
```
Comment (tachella) [inline examples/unfolded/demo_DEQ.py:7]: "visualizing" also double `` are missing in wandb_vis

File: examples/unfolded/demo_vanilla_unfolded.py [hunk 6]
```diff
@@ -1,9 +1,11 @@
...
-Unfolded algorithms for super-resolution
+Vanilla Unfolded algorithm for super-resolution
 ====================================================================================================
 
-This example shows you how to use unfolded architectures to solve a super-resolution problem.
-
+This is a simple example to show how to use vanilla unfolded Plug-and-Play.
+The DnCNN denoiser and the algorithm parameters (stepsize, regularization parameters) are trained jointly.
+For simplicity, we show how to train the algorithm on a  small dataset. For optimal results, use a larger dataset.
+For vializing the training, you can use Weight&Bias (wandb) by setting `wandb_vis=True`.
```
Comment (tachella) [inline examples/unfolded/demo_vanilla_unfolded.py:8]: same as before "visualizing" and double ``

**Review by tachella — CHANGES_REQUESTED** [review]
great PR! There's just a bunch of typos in the docs that I'll fix myself

**Pull request merged**